src/inmanta/deploy/state.py

```diff
@@ -283,13 +283,13 @@ def update_blocked_status(resource: ResourceIdStr) -> bool:
                 return False
             # Unblock resource
             self.resource_state[resource].blocked = BlockedStatus.NO
-            if self.resource_state[resource].status is ResourceStatus.HAS_UPDATE:
+            if self.resource_state[resource].status is ComplianceStatus.HAS_UPDATE:
```

Comment:
I think we may want to include `NON_COMPLIANT` here now? It's probably relatively rare that that would happen, but it is possible e.g. when a resource that was failing becomes blocked as a result of an (undefined) update to one of its dependencies.

Reply:
thx, completely missed that.